### ISSUE TITLE
Pluralization issue with certain languages

### ISSUE THREAD

**Issue by dangerdogz** [solidusio/solidus#1164]
Reposted from [Solidus I18](https://github.com/solidusio-contrib/solidus_i18n/issues/58)

References

[I18n Issue](https://github.com/svenfuchs/i18n/issues/239)
[ActiveAdmin Issue](https://github.com/activeadmin/activeadmin/pull/2791)
[Spree Issue](https://github.com/spree/spree/issues/5842)
[Spree Issue](https://github.com/spree/spree/pull/6059)

**Summary**

Basically, english use this function to generate plural names
[Rails i18n link](https://github.com/svenfuchs/rails-i18n/blob/master/lib/rails_i18n/common_pluralizations/one_other.rb)
/one_other.rb
```ruby
module RailsI18n
  module Pluralization
    module OneOther
      def self.rule
        lambda { |n| n == 1 ? :one : :other }
      end

      def self.with_locale(locale)
        { locale => {
            :'i18n' => {
              :plural => {
                :keys => [:one, :other],
                :rule => rule }}}}
      end
    end
  end
end
```

And some other languages like french use this one
[Rails i18n link](https://github.com/svenfuchs/rails-i18n/blob/master/lib/rails_i18n/common_pluralizations/one_upto_two_other.rb)
```ruby
module RailsI18n
  module Pluralization
    module OneUptoTwoOther
      def self.rule
        lambda { |n| n && n >= 0 && n < 2 ? :one : :other }
      end

      def self.with_locale(locale)
        { locale => {
            :'i18n' => {
              :plural => {
                :keys => [:one, :other],
                :rule => rule }}}}
      end
    end
  end
```

**Issue**

Throughout Solidus, the pluralization is triggered with this format :  ```<%= Spree::PromotionCategory.model_name.human(count: :many) %>```

It works with the english OneOther even if n is a symbol (:many) because :symbol == 1 will just return false (so you get the :other), but it crashes when you try to use another pluralization method because it's trying to determine if n ( a symbol, :many) is less or greater than an integer, thus you get :   ``` (comparison of Symbol with 0 failed)``` because you can't use this operator with a symbol

**Comment by dangerdogz** [comment]
Reposting comment by @tvdeyen :

The only way to fix this is to pass in an actual number, right? Good catch and great investigation @dangerdogz

We can use a number of the value of which depends on the language so it feels arbitrary to use something like 2.1 because it needs to be bigger than 2, and I haven't look into all the other calculators so there might be some other constraints.

**Comment by mamhoff** [comment]
As pointed out in https://github.com/spree/spree/issues/5842#issuecomment-68595785, the `ActiveAdmin` gem stored `2.1` in a constant to do the lookup properly. I'm not super-excited about that constant living in `I18n` in their solution, but I liked that it wasn't just a magic number. 

Maybe something like (in short): 
```
Spree::I18N_GENERIC_PLURAL = 2.1
Spree::Order.model_name.human(count: Spree::I18N_GENERIC_PLURAL) 
```

**Comment by dangerdogz** [comment]
I expect this is the better solution but it would imply editing tens or maybe hundreds of files so I was hoping for something else.

How would I go about updating this (in terms of PR size) ?

**Comment by tvdeyen** [comment]
:+1: for the ActiveAdmin approach. Thanks for finding this @mamhoff 

@dangerdogz I don't see any other chance in addressing this. You can split up the work into several chunks if you like or in one big, I don't mind. Since this is just a global search and replace, the review will be easy.

**Comment by dangerdogz** [comment]
@jhawthorn, @mamhoff  What do you guys think ? A big PR to replace everything at the same time, or smaller chunks?

**Comment by mamhoff** [comment]
As long as the commit does not mix concerns, it can be as big as you like, as far as I'm concerned. So one commit for introducing the thing, and another one that basically regexes over the entire project. That's probably even safer and more comprehensive than doing it bit by bit and missing one...

**Comment by dangerdogz** [comment]
Alright, going to make a PR tonight. Thanks for the feedback!

**Comment by dangerdogz** [comment]
Should be fixed by #1191
